src/ext/default.c

```diff
@@ -135,36 +169,40 @@ double DocScoreScorer(const ScoringFunctionArgs *ctx, const RSIndexResult *r,
  * DISMAX-style scorer
  *
  ******************************************************************************************/
-static double _dismaxRecursive(const RSIndexResult *r) {
+static double _dismaxRecursive(const ScoringFunctionArgs *ctx, const RSIndexResult *r) {
   // for terms - we return the term frequency
+  bool explain = ctx->scoreflags & SCORE_F_WITH_EXPLANATION;
   double ret = 0;
   switch (r->type) {
     case RSResultType_Term:
     case RSResultType_Numeric:
     case RSResultType_Virtual:
       ret = r->freq;
+      EXPLAIN(r, "DISMAX %.2f = Weight %.2f * Frequency %d", r->weight * ret, r->weight, r->freq);
       break;
     // for intersections - we sum up the term scores
     case RSResultType_Intersection:
       for (int i = 0; i < r->agg.numChildren; i++) {
-        ret += _dismaxRecursive(r->agg.children[i]);
+        ret += _dismaxRecursive(ctx, r->agg.children[i]);
       }
+      EXPLAIN(r, "%.2f = Weight %.2f * children DISMAX %.2f",  r->weight * ret, r->weight, ret);
       break;
     // for unions - we take the max frequency
     case RSResultType_Union:
       for (int i = 0; i < r->agg.numChildren; i++) {
-        ret = MAX(ret, _dismaxRecursive(r->agg.children[i]));
+        ret = MAX(ret, _dismaxRecursive(ctx, r->agg.children[i]));
```

Comment:
while we're at it, let's change the name of `_dismaxRecursive` and remove the underscore